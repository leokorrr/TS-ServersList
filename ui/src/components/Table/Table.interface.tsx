import { Server } from '../../types'

export default interface TableInterface {
    tableRows: Server[];
}