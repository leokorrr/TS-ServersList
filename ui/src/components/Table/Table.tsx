import React from 'react'
import TableRow from '../TableRow/TableRow'
import TableInterface from './Table.interface'
import TableHeader from '../TableHeader/TableHeader'

const Table: React.FC<TableInterface> = ({tableRows}) => {
    return (
        <table className="table">
            <thead>
                <TableHeader />
            </thead>
            <tbody>
                {tableRows.length === 0 && 
                    <tr className="table-row"><td className="table-row__cell">Loading...</td></tr>}
                {tableRows.map(tableRow => 
                    <TableRow key={tableRow.id} 
                        _id={tableRow.id} 
                        name={tableRow.name} 
                        status={tableRow.status}/>)}
            </tbody>
        </table>
    )
}

export default Table