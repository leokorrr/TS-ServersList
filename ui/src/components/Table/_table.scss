.table {
    border: 0.2rem solid $color-table-border;
    background-color: $color-white;
    width: 100%;
    border-collapse: collapse;
    margin-bottom: 2rem;
}