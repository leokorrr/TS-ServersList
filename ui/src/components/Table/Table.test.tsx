import React from 'react'
import Table from './Table'
import { shallow,  configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

const tableRows: [] = []
configure({ adapter: new Adapter() })
it("Table rendered without crashing", () => {
    shallow(<Table tableRows={tableRows}/>);
})
it("should render initial layout", () => {
    const component = shallow(<Table tableRows={tableRows}/>);
    expect(component.getElements()).toMatchSnapshot();
})