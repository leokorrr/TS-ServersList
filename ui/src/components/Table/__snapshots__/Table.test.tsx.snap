// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render initial layout 1`] = `
Array [
  <table
    className="table"
  >
    <thead>
      <TableHeader />
    </thead>
    <tbody>
      <tr
        className="table-row"
      >
        <td
          className="table-row__cell"
        >
          Loading...
        </td>
      </tr>
    </tbody>
  </table>,
]
`;
